debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -214,6 +214,11 @@ else if (!checkAgent) {
                     tablesSlot.set(getChangeTablesToQuery(partition, offsetContext, toLsn));
                     collectChangeTablesWithKnownStopLsn(partition, tablesSlot.get());
                 }
+
+                tablesSlot.set(Arrays.stream(tablesSlot.get())
+                        .filter(t -> !(t.getStopLsn().isAvailable() && t.getStopLsn().compareTo(fromLsn) <= 0))
```

Comment:
The condition could be simplified by removing parentheses and inverting all the inner conditions.